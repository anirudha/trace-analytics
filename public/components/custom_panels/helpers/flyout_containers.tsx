/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  EuiFlyout,
  EuiFlyoutBodyProps,
  EuiFlyoutFooterProps,
  EuiFlyoutHeaderProps,
  EuiOverlayMask,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';

type Props = {
  closeFlyout: () => void;
  flyoutHeader: JSX.Element;
  flyoutBody: JSX.Element;
  flyoutFooter: JSX.Element;
  ariaLabel: string;
};

export const FlyoutContainers = ({
  closeFlyout,
  flyoutHeader,
  flyoutBody,
  flyoutFooter,
  ariaLabel,
}: Props) => {
  return (
    <div>
      <EuiOverlayMask>
        <EuiFlyout ownFocus={false} onClose={() => closeFlyout()} aria-labelledby={ariaLabel}>
          {flyoutHeader}
          {flyoutBody}
          {flyoutFooter}
        </EuiFlyout>
      </EuiOverlayMask>
    </div>
  );
};
