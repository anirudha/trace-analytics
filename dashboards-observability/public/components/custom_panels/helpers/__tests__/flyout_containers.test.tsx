/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { FlyoutContainers } from '../flyout_containers';

describe('Flyout Container component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty flyout Container', () => {
    const closeFlyout = jest.fn();
    const flyoutHeader = <div />;
    const flyoutBody = <div />;
    const flyoutFooter = <div />;
    const ariaLabel = 'sampleFlyout';
    const wrapper = shallow(
      <FlyoutContainers
        closeFlyout={closeFlyout}
        flyoutHeader={flyoutHeader}
        flyoutBody={flyoutBody}
        flyoutFooter={flyoutFooter}
        ariaLabel={ariaLabel}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
