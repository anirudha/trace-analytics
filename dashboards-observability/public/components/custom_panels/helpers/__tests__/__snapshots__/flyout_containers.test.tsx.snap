// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout Container component renders empty flyout Container 1`] = `
<div>
  <EuiFlyout
    aria-labelledby="sampleFlyout"
    onClose={[Function]}
    ownFocus={false}
  >
    <div />
    <div />
    <div />
  </EuiFlyout>
</div>
`;
